subsys/pm/pm.c

```diff
@@ -27,7 +27,24 @@ LOG_MODULE_REGISTER(pm, CONFIG_PM_LOG_LEVEL);
 	(COND_CODE_1(CONFIG_SMP, (arch_curr_cpu()->id), (_current_cpu->id)))
 
 static ATOMIC_DEFINE(z_post_ops_required, CONFIG_MP_MAX_NUM_CPUS);
-static sys_slist_t pm_notifiers = SYS_SLIST_STATIC_INIT(&pm_notifiers);
+
+/* Active state notifiers list */
+static sys_slist_t pm_notifiers_active = SYS_SLIST_STATIC_INIT(&pm_notifiers_active);
+
+/* Generate a notifier list for each DT power state */
+DT_FOREACH_CHILD_STATUS_OKAY_SEP(DT_PATH(cpus, power_states), PM_NOTIFIERS_CREATE_STATES, (;));
```

Comment:
Should I wrap this in a `DT_NODE_EXISTS` check, in case the `cpus` node doesn't have a `power-states` child?

Reply:
Tested: DT errors will occur if `power-states` isn't defined. Adding `COND_CODE_1` wrapper